### PR TITLE
Add blocking of spindown when platform update is ongoing

### PR DESCRIPTION
**Pull request by jkcdarunday**

### CODE REVIEW

File: test/06-shutdown.test.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+test.hook('prepare throttled platform', async (t) => {
+  t.timeout(120_000)
+
+  const patchedArtefactDir = path.join(Helper.tmp, 'arcv-pear')
+  t.comment(`\tCopying platform code to ${patchedArtefactDir}`)
+  await fs.promises.mkdir(patchedArtefactDir, { recursive: true })
+  t.teardown(() => Helper.gc(patchedArtefactDir))
+  const mirror = new LocalDrive(rig.artefactDir).mirror(new LocalDrive(patchedArtefactDir), {
+    prune: false,
+    filter: (key) => !(key.startsWith('/pear/') || key.startsWith('/.git/') || key.startsWith('/test/'))
```
Comment (rafapaezbas) [inline test/06-shutdown.test.js:126]: use `ignore` instead of `filter`, will make the test run faster in windows

File: test/06-shutdown.test.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+test.hook('prepare low-spindown platform', async (t) => {
+  t.timeout(120_000)
+
+  const patchedArtefactDir = path.join(Helper.tmp, 'als-pear')
+  t.comment(`Copying platform code to ${patchedArtefactDir}`)
+  await fs.promises.mkdir(patchedArtefactDir, { recursive: true })
+  t.teardown(() => Helper.gc(patchedArtefactDir))
+  const mirror = new LocalDrive(rig.artefactDir).mirror(new LocalDrive(patchedArtefactDir), {
+    prune: false,
+    filter: (key) => !(key.startsWith('/pear/') || key.startsWith('/.git/') || key.startsWith('/test/'))
```
Comment (rafapaezbas) [inline test/06-shutdown.test.js:52]: same as below

File: test/06-shutdown.test.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  const hasSpunDown = await Promise.race([untilExit, timeoutUntil])
+  if (hasSpunDown === false) {
+    t.fail('sidecar failed to spin down')
+  } else {
+    t.pass('sidecar has spun down')
+  }
+})
+
+let platformDirRcv
+test.hook('prepare throttled platform', async (t) => {
```
Comment (rafapaezbas) [inline test/06-shutdown.test.js:117]: Do we need to mark this test as a hook?

File: test/06-shutdown.test.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  const dhtBootstrap = global.Pear.config.dhtBootstrap.map(e => `${e.host}:${e.port}`).join(',')
+  const sidecar = spawn(path.join(platformDirLs, 'current', BY_ARCH), ['--dht-bootstrap', dhtBootstrap, 'sidecar', '--key', staged.key], { stdio: 'pipe' })
+  t.teardown(() => { if (sidecar.exitCode === null) sidecar?.kill() })
+  const untilExit = new Promise(resolve => sidecar.once('exit', resolve))
+  t.teardown(async () => untilExit, { order: Infinity })
+
+  t.comment(`\tWaiting for sidecar spindown timeout to lapse (${(SPINDOWN_TIMEOUT + 10_000) / 1000}s)`)
+  const timeoutUntil = new Promise((resolve) => {
+    untilExit.finally(() => {
+      if (timeout) clearTimeout(timeout)
```
Comment (rafapaezbas) [inline test/06-shutdown.test.js:205]: where is timeout defined?

File: test/06-shutdown.test.js (hunk 3 above)
Comment (jkcdarunday) [reply test/06-shutdown.test.js:117]: @rafapaezbas Hmm.. I can merge this in the test where it's being used, but one issue is that the execution time of this step is inconsistent (last run was 30s on Linux and 60s on Windows, but I've seen this step go up to 90s). Right now, I have the timeouts for the spindown test a little tight but if I put this with the test, I'll have to loosen the timeout.

Would it be preferable to put this in the test instead?

**Pull request merged**